backend/custom_models/test.py

```diff
@@ -0,0 +1,19 @@
+from django.db import models
+from .item import Item
+MAX_CHAR_LEN = 100
+
+# Test Model (addional data for an item of type test)
+# Stores the test_name pk (a unique string to define the test), item id (parent item id),
+# is_public (is a sample test anyone can try out or not), default_time (the default time limit in minutes),
+# and type (string specifying the type of test)
+
+# NOTE: Test uses a String pk because migrations/rollbacks can result in different ids for tests
+# on a dev machine or a deployed environment. The string key will always be the same, no matter the environment
+
+
+class Test(models.Model):
+    test_name = models.CharField(primary_key=True, max_length=MAX_CHAR_LEN)
+    item_id = models.ForeignKey(Item, on_delete=models.CASCADE)

```

Comment:
Does this mean all items get deleted when this is deleted? Is that what we want?

Reply:
The opposite: if the item (the parent) is deleted, then this (the child) is also deleted
 "Django emulates the behavior of the SQL constraint ON DELETE CASCADE" (https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.CASCADE)

"[SQL] provides a more effective way called ON DELETE CASCADE referential action for a foreign key that allows you to delete data from child tables automatically when you delete the data from the parent table." (http://www.mysqltutorial.org/mysql-on-delete-cascade/ used this link since it has a visual example)